tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -96,6 +92,98 @@ internal readonly partial struct SecurityCoordinator
 
     internal static Span TryGetRoot(Span span) => span.Context.TraceContext?.RootSpan ?? span;
 
+    public IResult? RunWafForUser(string? userId = null, string? userLogin = null, string? userSessionId = null, bool fromSdk = false, Dictionary<string, string>? otherTags = null)
+    {
+        if (_httpTransport.IsBlocked)
+        {
+            return null;
+        }
+
+        IResult? result = null;
+        Dictionary<string, object>? addresses = null;
+        try
+        {
+            var additiveContext = GetOrCreateAdditiveContext();
+            if (additiveContext?.ShouldRunWith(_security, userId, userLogin, userSessionId, fromSdk) is { Count: > 0 } userAddresses)
+            {
+                addresses = userAddresses.ToDictionary(k => k.Key, object (v) => v.Value);
+
+                if (otherTags is not null)
+                {
+                    foreach (var kvp in otherTags)
+                    {
+                        if (!addresses.ContainsKey(kvp.Key))
+                        {
+                            addresses.Add(kvp.Key, kvp.Value);
+                        }
```

Comment:
Entirely, optional, but you could replace this with the following if you want to avoid the double-lookup
```suggestion
#if NETCOREAPP
                        addresses.TryAdd(kvp.Key, kvp.Value);
#else
                        if (!addresses.ContainsKey(kvp.Key))
                        {
                            addresses.Add(kvp.Key, kvp.Value);
                        }
#endif
```